### PR TITLE
tests: drivers: i2c: Add comprehensive I2C test validation suite

### PR DESCRIPTION
**Pull request by sudiiyengar-en**
Add extensive I2C driver test coverage including:

- Comprehensive validation tests (master mode, stress/robustness, hardware-software co-validation, regression tests)
- Boundary condition tests (size limits, addressing modes, speed transitions, error conditions)
- Fault injection tests (invalid addresses, NACK recovery, bus busy conditions, configuration edge cases)
- Hardware-aware tests (environmental stress, adaptive timing, signal integrity, power/temperature awareness)
- Performance tests (throughput, latency, concurrency, efficiency benchmarks, scalability analysis)
- Stress tests (write callback exactness, repeated start integrity)
- BME680 sensor integration test

Test infrastructure includes shared utilities, fault isolation, and data validation helpers across all test suites.

### CODE REVIEW

**Review by VeijoPesonen — DISMISSED** [review]

File: tests/drivers/i2c/src/test_i2c.c [hunk 1]
```diff
@@ -0,0 +1,985 @@
...
+		LOG_WRN("I2C target mode not supported");
+	} else {
+		zassert_equal(ret, 0,
+				  "i2c_target_unregister failed: %d", ret);
+	}
+
+	ctx->slave_registered = false;
+}
+
+static void i2c_teardown(void *fixture)
```
Comment (Shamshoor-AlifSemi) [inline tests/drivers/i2c/src/test_i2c.c]: after_fn is called after each unit test in this suite, then why are you again calling the same in the teardown_fn

File: tests/drivers/i2c/src/test_i2c.c [hunk 2]
```diff
@@ -0,0 +1,985 @@
...
+{
+	i2c_after(fixture);
+}
+
+ZTEST(i2c, test_node0)
+{
+	const struct device *const i2c_dev = DEVICE_DT_GET(I2C_MASTER);
```
Comment (Shamshoor-AlifSemi) [inline tests/drivers/i2c/src/test_i2c.c]: DEVICE_DT_GET(I2C_MASTER), DEVICE_DT_GET(I2C_SLAVE) are already resolved during startup_fn. Re-fetching it in each test case is redundant since the device pointer is static and invariant. Consider reusing the initialized handle to avoid duplication.

File: tests/drivers/i2c/src/test_i2c.h [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+#define I2C_MASTER DT_ALIAS(master_i2c)
+#define I2C_SLAVE  DT_ALIAS(slave_i2c)
+
+#define SLV_I2C_ADDR	  0x50
+#define SLV_I2C_10BITADDR 0x223
+
+#define BUFF_SIZE		 4
+#define BUFF_PERF		 128
+#define SLAVE_T_MASTER_R 1
```
Comment (Saba-AlifSemi) [inline tests/drivers/i2c/src/test_i2c.h]: Hardcoding this as a macro limits flexibility. Since this is a configurable behavior, it should be exposed via Kconfig (CONFIG_SLAVE_T_MASTER_R) instead of #define, enabling build-time configuration without modifying source code.

File: tests/drivers/i2c/src/test_i2c_boundary.c [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+ * @brief Test FIFO boundary conditions and overflow detection
+ *
+ * Objective: Expose FIFO overflow/underflow bugs in the driver
+ * Hardware Issue: FIFO depth mismatches, pointer corruption
+ * Driver Issue: Missing FIFO full/empty checks
+ */
+ZTEST(i2c_boundary_suite, test_fifo_boundary)
+{
+	TC_PRINT("Running boundary test: FIFO boundary\n");
+	const struct device *master = DEVICE_DT_GET(I2C_MASTER);
```
Comment (Shamshoor-AlifSemi) [inline tests/drivers/i2c/src/test_i2c_boundary.c]: DEVICE_DT_GET(I2C_MASTER) is already resolved during startup_fn. Re-fetching it in each test case is redundant since the device pointer is static and invariant. Consider reusing the initialized handle to avoid duplication.

File: tests/drivers/i2c/src/test_i2c_comprehensive.c [hunk 5]
```diff
@@ -0,0 +1,1096 @@
...
+}
+
+/**
+ * @brief Comprehensive stress and robustness validation
+ *
+ * Objective: Stress test I2C driver and hardware to breaking point
+ * Hardware Issue: Signal integrity under stress, thermal issues
+ * Driver Issue: Memory leaks, state corruption, resource exhaustion
+ */
+ZTEST(i2c_comprehensive_suite, test_comprehensive_stress)
```
Comment (Shamshoor-AlifSemi) [inline tests/drivers/i2c/src/test_i2c_comprehensive.c]: isn't the same test covered in stress testcases?

File: tests/drivers/i2c/src/test_i2c_bme680.c [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+		ret = sensor_channel_get(dev, SENSOR_CHAN_GAS_RES, &gas_res);
+		zassert_ok(ret, "Failed to read gas resistance: %d", ret);
+		LOG_INF("T: %d.%06d; P: %d.%06d; H: %d.%06d; G: %d.%06d",
+			temp.val1, temp.val2, press.val1, press.val2, humidity.val1,
+			humidity.val2, gas_res.val1, gas_res.val2);
+		i++;
+	}
+}
+
+static void *i2c_sensor_setup(void)
```
Comment (Saba-AlifSemi) [inline tests/drivers/i2c/src/test_i2c_bme680.c]: Instead of leaving the test hooks empty, consider using setup to initialize the device (e.g., DEVICE_DT_GET_ONE and device_is_ready) and pass it via a fixture. This will improve structure and reduce duplication in test cases.

File: tests/drivers/i2c/src/test_i2c_bme680.c [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+{
+	return NULL;
+}
+
+static void i2c_sensor_before(void *fixture)
+{
+	ARG_UNUSED(fixture);
```
Comment (Saba-AlifSemi) [inline tests/drivers/i2c/src/test_i2c_bme680.c]: before: Useful for per-test initialization, such as resetting sensor state or ensuring a fresh sample cycle before each test case.

File: tests/drivers/i2c/src/test_i2c_bme680.c [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+static void i2c_sensor_after(void *fixture)
+{
+	ARG_UNUSED(fixture);
```
Comment (Saba-AlifSemi) [inline tests/drivers/i2c/src/test_i2c_bme680.c]: after: Can handle post-test cleanup or logging, especially if future tests modify sensor configuration or state.

File: tests/drivers/i2c/README.rst [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+- ``CONFIG_I2C_ENHANCED_STRESS``  Extended stress with fault injection
+- ``CONFIG_I2C_COMPREHENSIVE_VALIDATION`` Legacy aggregation
+
+Frequency selection (multi-frequency execution):
+
+- ``CONFIG_I2C_TEST_FREQ_STANDARD``   Include Standard mode (100 kHz) in test loops (default: y)
+- ``CONFIG_I2C_TEST_FREQ_FAST``       Include Fast mode (400 kHz) in test loops (default: y)
+- ``CONFIG_I2C_TEST_FREQ_FAST_PLUS``  Include Fast Plus mode (1 MHz) in test loops (default: y)
+- ``CONFIG_I2C_TEST_FREQ_HIGH``       Include High Speed mode (3.4 MHz) in test loops (default: y)
+
```
Comment (Saba-AlifSemi) [inline tests/drivers/i2c/README.rst:33]: can you add the missing Kconfigs

File: tests/drivers/i2c/src/test_i2c.c [hunk 10]
```diff
@@ -0,0 +1,985 @@
...
+{
+	test_start_i2c_transmit();
+}
+
+ZTEST(i2c, test_xcv)
+{
+	test_start_i2c_transceive();
+}
+
+ZTEST(i2c, test_nack_recovery)
```
Comment (Shamshoor-AlifSemi) [inline tests/drivers/i2c/src/test_i2c.c]: The basic NACK recovery flow tested here appears to be a subset of the more extensive fault-injection coverage in test_i2c_fault_injection.c:test_nack_recovery. It may be worth evaluating whether this can be consolidated to reduce duplication, while still preserving frequency-specific validation.

File: tests/drivers/i2c/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,118 @@
+cmake_minimum_required(VERSION 3.20.0)
+
+# ==================== ZEPHYR PROJECT SETUP ==================
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(alif_i2c)
+
+# ==================== TEST CONFIGURATION OPTIONS ============
+# Enable BME680 sensor tests (mutually exclusive with others)
+option(BME680 "Enable BME680 sensor integration tests" OFF)
```
Comment (Saba-AlifSemi) [inline tests/drivers/i2c/CMakeLists.txt]: The test selection is currently controlled via a CMake option (option(BME680 ...)), which is not aligned with Zephyr’s standard configuration approach. Since this is a feature/test toggle, it would be better to use Kconfig instead.

File: tests/drivers/i2c/src/test_i2c_fault_injection.c [hunk 12]
```diff
@@ -0,0 +1,613 @@
...
+	}
+}
+
+/**
+ * @brief Test NACK recovery scenarios
+ *
+ * Objective: Validate driver recovery after NACK conditions
+ * Expected: Driver should recover and handle subsequent transfers
+ */
+ZTEST(i2c_fault_suite, test_nack_recovery)
```
Comment (Shamshoor-AlifSemi) [inline tests/drivers/i2c/src/test_i2c_fault_injection.c]: Similar test is observed in test_i2c.c also, can those two be integrated and kept as single testcase to avoid redundancy.

File: tests/drivers/i2c/src/test_i2c_performance.c [hunk 13]
```diff
@@ -0,0 +1,663 @@
...
+	if (ret == -ENOTSUP) {
+		LOG_WRN("I2C target mode not supported");
+	} else {
+		zassert_equal(ret, 0, "i2c_target_unregister failed: %d", ret);
+	}
+
+	ctx->slave_registered = false;
+}
+
+static void i2c_perf_teardown(void *fixture)
```
Comment (Shamshoor-AlifSemi) [inline tests/drivers/i2c/src/test_i2c_performance.c]: after_fn is called after each unit test in this suite, then why are you again calling the same in the teardown_fn

**Review by Shamshoor-AlifSemi — CHANGES_REQUESTED** [review]
some changes are required.

File: tests/drivers/i2c/CMakeLists.txt (hunk 11 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/CMakeLists.txt]: Thanks, updated.

File: tests/drivers/i2c/src/test_i2c.c (hunk 1 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c.c]: Thanks, updated.

File: tests/drivers/i2c/src/test_i2c.c (hunk 2 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c.c]: Thanks, updated.

File: tests/drivers/i2c/src/test_i2c_boundary.c (hunk 4 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_boundary.c]: Thanks, updated.

File: tests/drivers/i2c/src/test_i2c_comprehensive.c (hunk 5 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_comprehensive.c]: Thanks, Updated.

File: tests/drivers/i2c/src/test_i2c.c (hunk 10 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c.c]: test_i2c.c version: Hardware capability validation - ensures NACK recovery works across Standard→Fast→Fast+→High speeds. This catches speed-specific hardware bugs.
fault_injection version: Diagnostic tooling - when recovery fails, tests multiple strategies (immediate, delayed, reset) to identify root cause.

File: tests/drivers/i2c/src/test_i2c_fault_injection.c (hunk 12 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_fault_injection.c]: test_i2c.c version: Hardware capability validation - ensures NACK recovery works across Standard→Fast→Fast+→High speeds. This catches speed-specific hardware bugs.
fault_injection version: Diagnostic tooling - when recovery fails, tests multiple strategies (immediate, delayed, reset) to identify root cause.

File: tests/drivers/i2c/src/test_i2c_performance.c (hunk 13 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_performance.c]: Thanks, updated.

File: tests/drivers/i2c/README.rst (hunk 9 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/README.rst:33]: Thanks, updated.

File: tests/drivers/i2c/src/test_i2c_bme680.c (hunk 8 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_bme680.c]: Thanks, updated.

File: tests/drivers/i2c/src/test_i2c_bme680.c (hunk 7 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_bme680.c]: Thanks, updated.

File: tests/drivers/i2c/src/test_i2c_bme680.c (hunk 6 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_bme680.c]: Thanks, updated.

File: tests/drivers/i2c/src/test_i2c.h (hunk 3 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c.h]: Thanks, updated.

File: tests/drivers/i2c/README.rst [hunk 14]
```diff
@@ -0,0 +1,292 @@
...
+- Payload truncation possible
+- Uses ``buf_write_received``
+- Batch processing affects timing
+
+Recommended usage:
+
+Performance:
+
+    west build ... -DCONFIG_I2C_PERFORMANCE_TESTS=y
```
Comment (VeijoPesonen) [inline tests/drivers/i2c/README.rst]: @sudiiyengar-en Please check our User Guide for proper syntax. @Anilkumar-Alifsemi would you please assist here how things should be done.
```suggestion
.. code-block:: console
    west build ... -DCONFIG_I2C_PERFORMANCE_TESTS=y
```

File: tests/drivers/i2c/README.rst [hunk 15]
```diff
@@ -0,0 +1,292 @@
...
+
+    west build ... \
+      -DCONFIG_I2C_TARGET_BUFFER_MODE=y \
+      -DCONFIG_I2C_STRESS_TESTS=y \
+      -DCONFIG_I2C_FAULT_INJECTION_TESTS=y
+
+Test compatibility
+------------------
+
+| Suite              | Buffer mode | Status |
```
Comment (VeijoPesonen) [inline tests/drivers/i2c/README.rst]: Wrong syntax for tables.

**Review by VeijoPesonen — CHANGES_REQUESTED** [review]
@sudiiyengar-en Please seek advice from @Anilkumar-Alifsemi how the documentation should be written.

File: tests/drivers/i2c/README.rst (hunk 14 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/README.rst]: Thanks, updated.

File: tests/drivers/i2c/README.rst (hunk 15 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/README.rst]: Thanks, Updated .

File: tests/drivers/i2c/README.rst [hunk 16]
```diff
@@ -0,0 +1,307 @@
...
+- Uses ``buf_write_received``
+- Batch processing affects timing
+
+Recommended usage:
+
+Performance:
+
+    west build -p always -b <board> \
+      tests/drivers/i2c \
+      -DCONFIG_I2C_PERFORMANCE_TESTS=y
```
Comment (VeijoPesonen) [inline tests/drivers/i2c/README.rst]:
```suggestion
.. code-block:: console

    west build -p always -b <board> \
      tests/drivers/i2c \
      -DCONFIG_I2C_PERFORMANCE_TESTS=y
```

Applies for the other similar cases too.

Comment (VeijoPesonen) [reply tests/drivers/i2c/README.rst]: @Anilkumar-Alifsemi Please review.

Comment (sudiiyengar-en) [reply tests/drivers/i2c/README.rst]: updated

**Review by VeijoPesonen — DISMISSED** [review]

**Review by Anilkumar-Alifsemi — DISMISSED** [review]
README.rst looks fine for me

**Review by VeijoPesonen — DISMISSED** [review]

File: tests/drivers/i2c/boards/i2c.overlay [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+/ {
+	aliases {
+		master-i2c = &i2c0;
+		slave-i2c = &i2c1;
+	};
+};
+
+&i2c0 {
+	status = "okay";
+	/* Master controller for restart condition validation */
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/boards/i2c.overlay]: Please don't use master & slave as people might find it offensive. Use e.g. controller/target instead. This applies to all files

File: tests/drivers/i2c/Kconfig [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+#License Agreement with this file.If not, please write to:
+#contact @alifsemi.com, or visit : https: // alifsemi.com/license
+
+menu "Alif I2C test options"
+
+#-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -
+#Comprehensive test selection
+#-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -
+config I2C_ALL_TESTS
+bool "Enable all I2C test suites"
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: Fix indentation

File: tests/drivers/i2c/Kconfig [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+		Include High Speed mode (3.4 MHz) in multi-frequency test loops.
+		When enabled, transfer-bearing tests will execute at this speed.
+		Most hardware does not support 3.4 MHz; disable via extra_configs
+		or board configuration if unsupported.
+
+#-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -
+#Stress testing options
+#-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -
+config I2C_STRESS_TESTS
+bool "Enable stress testing suite"
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: indentation

File: tests/drivers/i2c/Kconfig [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+help
+	Enable basic stress testing with restart condition validation
+	and callback integrity checks. This provides comprehensive stress
+	validation without automatic error injection or enhanced timing
+	validation. For complete stress testing with bug detection,
+	enable I2C_ENHANCED_STRESS instead.
+
+config I2C_ENHANCED_STRESS
+bool "Enable enhanced stress testing with bug detection"
+select I2C_ERROR_INJECTION
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: indentation

File: tests/drivers/i2c/Kconfig [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+	help
+		Enable deliberate timing perturbations during the enhanced
+		stress suite. This symbol is auto-selected by
+		CONFIG_I2C_ENHANCED_STRESS to preserve the current test
+		behavior while removing Kconfig warnings for an undefined
+		symbol.
+
+config I2C_STRESS_LOOPS
+int "Number of stress test iterations"
+depends on I2C_ENHANCED_STRESS
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: indentation

File: tests/drivers/i2c/Kconfig [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+default 1000
+help
+	Number of iterations for stress tests. Higher values provide
+	more thorough validation but increase test duration.
+
+config I2C_STRESS_TIMEOUT
+int "Stress test timeout in milliseconds"
+depends on I2C_ENHANCED_STRESS
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: indentation

File: tests/drivers/i2c/Kconfig [hunk 23]
```diff
@@ -0,0 +1,169 @@
...
+help
+	Maximum time allowed for stress test completion. Tests will
+	fail if they exceed this timeout.
+
+#-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -
+#Individual test suite selection
+#-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -
+config I2C_PERFORMANCE_TESTS
+bool "Enable performance testing suite"
+default n
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: indentation

File: tests/drivers/i2c/Kconfig [hunk 24]
```diff
@@ -0,0 +1,169 @@
...
+help
+	Enable performance tests that validate I2C throughput and
+	latency at different bus speeds (standard, fast, fast-plus, high).
+	When enabled, this runs the complete performance test suite with
+	all speed configurations.
+
+config I2C_BOUNDARY_TESTS
+bool "Enable boundary condition testing"
+default n
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: indentation

File: tests/drivers/i2c/Kconfig [hunk 25]
```diff
@@ -0,0 +1,169 @@
...
+help
+	Enable boundary condition tests that validate edge cases,
+	maximum sizes, and critical data patterns. When enabled,
+	this runs FIFO, register configuration, timing, and data pattern
+	boundary tests.
+
+config I2C_HARDWARE_AWARE_TESTS
+bool "Enable hardware-specific testing"
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: indentation

File: tests/drivers/i2c/Kconfig [hunk 26]
```diff
@@ -0,0 +1,169 @@
...
+default n
+help
+	Enable hardware-aware tests that validate platform-specific
+	behavior and timing characteristics. When enabled, this runs
+	interrupt handling, clock stretching, signal integrity, and DMA
+	interaction tests.
+
+config I2C_FAULT_INJECTION_TESTS
+bool "Enable fault injection testing"
+default n
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: indentation

File: tests/drivers/i2c/Kconfig [hunk 27]
```diff
@@ -0,0 +1,169 @@
...
+	and recovery mechanisms under various failure conditions.
+	When enabled, this runs the fault injection test suite.
+
+	Note: This is independent from I2C_ENHANCED_STRESS.
+	I2C_FAULT_INJECTION_TESTS focuses on specific fault scenarios,
+	while I2C_ENHANCED_STRESS provides comprehensive stress testing.
+
+config I2C_BME680_TESTS
+bool "Enable BME680 sensor integration tests"
+default n
```
Comment (AnttiKauppila-AlifSemi) [inline tests/drivers/i2c/Kconfig]: indentation

**Review by AnttiKauppila-AlifSemi — CHANGES_REQUESTED** [review]
Wording should be changed (replace master & slave with more appropriate names)
KConfig indentations needs to be fixed

File: tests/drivers/i2c/boards/i2c.overlay (hunk 17 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/boards/i2c.overlay]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 18 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 19 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 20 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 21 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 22 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 23 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 24 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 25 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 26 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

File: tests/drivers/i2c/Kconfig (hunk 27 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/Kconfig]: Thanks, Updated.

**Review by VeijoPesonen — DISMISSED** [review]

**Review by Saba-AlifSemi — COMMENTED** [review]
looks  good

**Review by Shamshoor-AlifSemi — COMMENTED** [review]
looks good!

**Review by Shamshoor-AlifSemi — DISMISSED** [review]

**Review by Saba-AlifSemi — DISMISSED** [review]
looks good

File: tests/drivers/i2c/boards/i2c.overlay
Comment (VeijoPesonen) [inline tests/drivers/i2c/boards/i2c.overlay:1]: @sudiiyengar-en  You should be using "controller/target" instead of "master/slave" .

Comment (sudiiyengar-en) [reply tests/drivers/i2c/boards/i2c.overlay:1]: Updated, Thanks for the review.

File: tests/drivers/i2c/src/test_i2c_performance.c [hunk 28]
```diff
@@ -0,0 +1,516 @@
...
+						     uint8_t val)
+{
+	ARG_UNUSED(config);
+	if (i2c_test_ctx.tgt_rx_cnt < BUFF_PERF) {
+		i2c_test_ctx.tgt_rx_data[i2c_test_ctx.tgt_rx_cnt++] = val;
+		LOG_DBG("Target RX [%u]: 0x%02X (total: %u)",
+			i2c_test_ctx.tgt_rx_cnt - 1, val,
+			i2c_test_ctx.tgt_rx_cnt);
+	} else {
+		LOG_DBG("Ignoring extra slave RX byte 0x%02X", val);
```
Comment (VeijoPesonen) [inline tests/drivers/i2c/src/test_i2c_performance.c]: Controller&Target

File: tests/drivers/i2c/src/test_i2c_performance.c [hunk 29]
```diff
@@ -0,0 +1,516 @@
...
+
+	LOG_INF("Received %d bytes from controller: ", len);
+	for (idx = 0; idx < len; idx++) {
+		/* Bounds checking to prevent buffer overflow */
+		if (i2c_test_ctx.tgt_rx_cnt < BUFF_PERF) {
+			i2c_test_ctx.tgt_rx_data[i2c_test_ctx.tgt_rx_cnt++] =
+				data_buf[idx];
+			LOG_INF("0x%x ", data_buf[idx]);
+		} else {
+			LOG_DBG("Ignoring extra slave RX byte 0x%x",
```
Comment (VeijoPesonen) [inline tests/drivers/i2c/src/test_i2c_performance.c]: Controller&Target

File: tests/drivers/i2c/src/test_i2c_performance.c [hunk 30]
```diff
@@ -0,0 +1,516 @@
...
+	i2c_test_ctx.tgt_rx_cnt = 0;
+	i2c_test_ctx.tgt_tx_cnt = 0;
+	memset(i2c_test_ctx.tgt_rx_data, 0, BUFF_PERF);
+	memset(i2c_test_ctx.tgt_tx_data, 0, BUFF_PERF);
+
+	i2c_tcfg_perf.flags = 0;
+	i2c_tcfg_perf.address = TGT_I2C_ADDR;
+
+	ret = i2c_configure(i2c_target_dev, i2c_cfg);
+	zassert_equal(ret, 0, "I2C slave config failed");
```
Comment (VeijoPesonen) [inline tests/drivers/i2c/src/test_i2c_performance.c]: Controller&Target

File: tests/drivers/i2c/src/test_i2c_performance.c [hunk 31]
```diff
@@ -0,0 +1,516 @@
...
+	size_t transfer_len = BUFF_PERF;
+	const struct device *const i2c_controller_dev = i2c_test_ctx.controller_dev;
+
+	ret = i2c_configure(i2c_controller_dev, i2c_test_ctx.i2c_cfg);
+	if (ret != 0) {
+		LOG_ERR("I2C controller configuration failed: %d", ret);
+		return ret;
+	}
+
+	/* Use distinct random patterns: controller TX != slave TX */
```
Comment (VeijoPesonen) [inline tests/drivers/i2c/src/test_i2c_performance.c]: Controller&Target

File: tests/drivers/i2c/src/test_i2c_performance.c (hunk 28 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_performance.c]: Updated

File: tests/drivers/i2c/src/test_i2c_performance.c (hunk 29 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_performance.c]: Updated

File: tests/drivers/i2c/src/test_i2c_performance.c (hunk 31 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_performance.c]: Updated

File: tests/drivers/i2c/src/test_i2c_performance.c (hunk 30 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_performance.c]: Updated

**Review by VeijoPesonen — DISMISSED** [review]

File: tests/drivers/i2c/boards/lpi2c_dw.conf [hunk 32]
```diff
@@ -0,0 +1,3 @@
+# DesignWare I2C configuration for LPI2C1 tests
+CONFIG_I2C_DW=y
+CONFIG_I2C_DW_CLOCK_SPEED=160
```
Comment (petrih-alifsemi) [inline tests/drivers/i2c/boards/lpi2c_dw.conf]: Why is this have to be defined in conf overlay? This should be defined in `defconfig` by default per SoC, right?

Comment (petrih-alifsemi) [reply tests/drivers/i2c/boards/lpi2c_dw.conf]: ...actually the clock freq should be read from registers to make sure it is 100% correct. Currently I2S driver uses fixed values which will conflict with the aiPM configurations...

Comment (sudiiyengar-en) [reply tests/drivers/i2c/boards/lpi2c_dw.conf]: https://github.com/alifsemi/zephyr_alif/pull/464. This PR is still in review. Before this, I took the implementation as a reference.

**Review by VeijoPesonen — DISMISSED** [review]

File: tests/drivers/i2c/boards/i2c.overlay
Comment (VeijoPesonen) [reply tests/drivers/i2c/boards/i2c.overlay:1]: A regression. Please fix again

Comment (VeijoPesonen) [reply tests/drivers/i2c/boards/i2c.overlay:1]: Check all the other files too.

**Review by VeijoPesonen — CHANGES_REQUESTED** [review]
Still naming issues.

**Comment by sudiiyengar-en** [comment]
> Still naming issues.

Sorry , I will correct it

**Review by VeijoPesonen — DISMISSED** [review]

**Review by RupeshKumar-AlifSemi — DISMISSED** [review]

**Review by AnttiKauppila-AlifSemi — APPROVED** [review]

**Review by VeijoPesonen — APPROVED** [review]
